src/private/WRLDRouteView.mm

```diff
@@ -0,0 +1,301 @@
+                }
+            }
+            else
+            {
+                if(isActiveStep)
+                {
+                    [self addLineCreationParamsForStep:step stepIndex:flattenStepIndex closestPointOnPath:closestPointOnRoute splitIndex:indexOfPathSegmentStartVertex];
+                }
+                else
+                {
+                    [self addLineCreationParamsForStep:step ndFlattenStepIndex:flattenStepIndex];
+                    
+                }
+            }
+            flattenStepIndex ++;
+        }
+    }
+    [self refreshPolylines];
+}
+
+-(void) addLineCreationParamsForStep:(WRLDRouteStep*)routeStep
+                          stepBefore:(WRLDRouteStep*)routeStepBefore
+                           stepAfter:(WRLDRouteStep*)routeStepAfter
+                    flattenStepIndex:(int)flattenStepIndex
+                               color:(UIColor *)color
+{
+    if (routeStep.pathCount < 2)
+    {
+        return;
+    }
+    
+    m_routeStepToPolylineCreateParams[flattenStepIndex] = [WRLDRouteViewHelper CreateLinesForFloorTransition:routeStep
+                                                                                                 floorBefore:routeStepBefore.indoorFloorId
+                                                                                                  floorAfter:routeStepAfter.indoorFloorId
+                                                                                                     color:color];
+}
+
+-(void) addLineCreationParamsForStep:(WRLDRouteStep*)routeStep ndFlattenStepIndex:(int)flattenStepIndex
+{
+    if (routeStep.pathCount < 2)
+    {
+        return;
+    }
+    
+    m_routeStepToPolylineCreateParams[flattenStepIndex] = [WRLDRouteViewHelper CreateLinesForRouteDirection:routeStep color:m_color];
+}
+
+-(void) addLineCreationParamsForStep:(WRLDRouteStep*)routeStep
+                           stepIndex:(int)stepIndex
+                  closestPointOnPath:(CLLocationCoordinate2D)closestPointOnPath
+                        splitIndex:(int)splitIndex
+{
+    if (routeStep.pathCount < 2)
+    {
+        return;
+    }
+
+    m_routeStepToPolylineCreateParams[stepIndex] = [WRLDRouteViewHelper CreateLinesForRouteDirection:routeStep forwardColor:m_forwardPathColor backwardColor:m_color splitIndex:splitIndex closestPointOnPath:closestPointOnPath];
+}
+
+-(void) refreshPolylines
+{
+    [self removeFromMap];
+    
+    Eegeo_ASSERT(m_polylineIds.count == 0);
+
+    WRLDRoutingPolylineCreateParamsVector allPolylineCreateParams;
+    
+    for (const auto& pair : m_routeStepToPolylineCreateParams)
+    {
+        const auto& createParams = pair.second;
+        allPolylineCreateParams.insert(allPolylineCreateParams.end(), createParams.begin(), createParams.end());
+    }
+    
+    m_polylineIds = [WRLDRouteViewAmalgamationHelper CreatePolylines:allPolylineCreateParams width:m_width miterLimit:m_miterLimit];
```

Comment:
I think this is returning polylines instead of polyline ids so can rename the variable here.